dali/python/nvidia/dali/auto_aug/core/decorator.py

```diff
@@ -0,0 +1,100 @@
+        "Could not import numpy. DALI's automatic augmentation examples depend on numpy. "
+        "Please install numpy to use the examples.")
+
+from numpy import typing as npt
+
+
+def augmentation(function: Optional[Callable[..., _DataNode]] = None, *,
+                 mag_range: Optional[Union[Tuple[float, float], np.ndarray]] = None,
+                 randomly_negate: Optional[bool] = None,
+                 as_param: Optional[Callable[[float], npt.ArrayLike]] = None,
+                 param_device: Optional[str] = None, name: Optional[str] = None,
+                 augmentation_cls: Type[Augmentation] = None):
+    """
+    A decorator turning transformations implemented with DALI into augmentations that
+    can be used by automatic augmentations (e.g. AutoAugment, RandAugment, TrivialAugment).
+
+    The `function` must accept at least two args: a sample and a parameter.
+    The decorator handles computation of the parameter. Instead of the parameter, the
+    decorated augmentation accepts magnitude bin and the total number of bins.
+    Then, the bin is used to compute the parameter as if by calling
+    `as_param(magnitudes[magnitude_bin] * ((-1) ** random_sign))`, where
+    `magnitudes=linspace(mag_range[0], mag_range[1], num_magnitude_bins)`.
+
+    Parameter
+    ---------
+    function : callable
+        A function that accepts at least two positional args: a batch
+        (represented as DataNode) to be processed, and a parameter of the transformation.
+        The function must use DALI operators to process the batch and return a single output
+        of such processing.
+    mag_range : (number, number) or np.ndarray
+        Specifies the range of applicable magnitudes for the operation.
+        If the tuple is provided, the magnitudes will be computed as
+        `linspace(mag_range[0], mag_range[1], num_magnitude_bins)`.
+        If the np.ndarray is provided, it will be used directly instead of the linspace.
+        If no `mag_range` is specified, the parameter passed to the `function` will be `None`.
+    randomly_negate: bool
+        If `True`, the magnitude from the mag_range will be randomly negated for every sample.
+    as_param: callable
+        A callback that transforms the magnitude into a parameter. The parameter will be passed to
+        the decorated operation instead of the plain magnitude. This way, the parameters for the
+        range of magnitudes can be computed once in advance and stored as a Constant node.
+        Note, the callback must return numpy arrays or data directly convertible to numpy arrays
+        (in particular, no pipelined DALI operators can be used in the callback).
+        The output type and dimensionality must be consistent and not depend on the magnitude.
+    param_device: str
+        A "cpu" or "gpu", describes where to store the precomputed parameters.
+    name: str
+        Name of the augmentation. By default, the name of the decorated function is used.
```

Comment:
I don't see the name of the function accessed anywhere, or maybe I missed it somehow.

Reply:
```
@property
    def name(self):
        return self._name or self.op.__name__
```